src/activities/assignments/AssignmentEntity.js

```diff
@@ -816,6 +888,13 @@ export class AssignmentEntity extends Entity {
 			fields.push({ name: 'submissionType', value: assignment.submissionType });
 		}
 
+		const shouldSaveAllowableFileType = typeof assignment.allowableFileType !== 'undefined' &&
```

Comment:
Might just be me, but I find this hard to read. Perhaps break it out into smaller variables, and then use those variables to determine the value of `shouldSaveAllowableFileType`

Would also make debugging easier :)